src/main/webapp/image-upload-validation.js

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// suppress linter error -
+// validateImageUpload function is used from another file.
+/* eslint no-unused-vars:
+["error", { "varsIgnorePattern": "validateImageUpload" }] */
+
+'use strict';
+
+/**
+ * Function to validate the uploaded image.
+ * If valid - returns nothing, if not - throws error.
+ * @throws {Error} validation errors
+ */
+async function validateImageUpload() {
+  // get FileList with all the files from input element.
+  const files = document.getElementById('upload-image').files;
+
+  // check if no files were uploaded.
+  if (files.length === 0) {
+    throw new Error('Nothing is uploaded');
+  }
+
+  // get first (and the only) file from FileList.
+  const file = files[0];
+
+  // check whether file is jpeg or png.
+  const fileType = await getImageTypeOrError(file);
+
+  // check file size.
+  validateImageSize(file, fileType);
+
+  // Image resolution can not be more than 1920x1080.
+  await validateImageResolution(file, 1920, 1080);
+}
+
+/**
+ * Function to make sure the image is not too big.
+ * @param {File} image
+ * @param {string} imageType Can be 'png' or 'jpeg' only.
+ */
+function validateImageSize(image, imageType) {
+  // This limit comes from simple calculation.
+  // MAX_RESOLUTION * AVERAGE_NUMBER_OF_BYTES_PER_PIXEL_IN_PNG
+  // 1920 x 1080 x 4 ~ 8294400 bytes ~ 8 mb
+  const PNG_LIMIT_MB = 8;
```

Comment:
Also I think this could be calculated using the constants that I suggested above.